File: Examen/Examen_1er_Bimestre/src/com/ejemplo/controller/AutorController.java
package com.ejemplo.controller;

import java.com.ejemplo.model.Autor;
import java.com.ejemplo.service.AutorService;

import java.io.IOException;
import java.util.List;

public class AutorController {
    private AutorService autorService = new AutorService();

    // Crear un nuevo autor
    public void createAutor(Autor autor) throws IOException {
        autorService.saveAutor(autor);
    }

    // Obtener todos los autores
    public List<Autor> getAutores() throws IOException {
        return autorService.getAllAutores();
    }

    // Actualizar un autor existente
    public void updateAutor(Autor autor) throws IOException {
        autorService.updateAutor(autor);
    }

    // Eliminar un autor por ID
    public void deleteAutor(int id) throws IOException {
        autorService.deleteAutor(id);
    }

    // Obtener un autor por ID
    public Autor getAutorById(int id) throws IOException {
        return autorService.getAutorById(id);
    }
}

File: Examen_02/src/controller/AutorController.kt
package controller

import dao.AutorDao
import model.Autor
import java.io.IOException


class AutorController {
    private val autorDao = AutorDao()

    @Throws(IOException::class)
    fun createAutor(autor: Autor) {
        autorDao.save(autor)
    }

    @Throws(IOException::class)
    fun getAutores(): List<Autor> {
        return autorDao.findAll()
    }

    @Throws(IOException::class)
    fun updateAutor(autor: Autor) {
        autorDao.update(autor)
    }

    @Throws(IOException::class)
    fun deleteAutor(id: Int) {
        autorDao.delete(id)
    }

    @Throws(IOException::class)
    fun getAutorById(id: Int): Autor? {
        return autorDao.findById(id)
    }
}
